### DISCUSSION TITLE
save patches on Cardinal

### DISCUSSION THREAD

**Discussion by Marco-56** [DISTRHO/Cardinal#608] [Q&A]
I an new on Cardinal, I use the standalone version and it's very interessant, but I found frustrating that the "save" function doesn't works; I mean: if I make changes on a patch (from the repository f.ex.) and then press save, when I reopen the patch the changes are lost! The same happens also on the VST version (inside Waveform). Also if I change name and use the "save as" command (creating a new patch) the changes are lost! Someone can tell me where I make mistakes?

**Comment by dromer** [comment]
On which OS and plugin version is this?

Save function most definitely works, just look at all the patches saved and shared on Patchstorage: https://patchstorage.com/platform/cardinal/

**Reply by Marco-56** [reply]
OS Windows 11, and 23.10 Cardinal. But the mostly "save" doesn't works.

**Reply by Marco-56** [reply]
I also tried to launch Cardinal as administrator, but "save" doesn't works (in fact sometimes, but sometimes) it works. The issue looks evident becouse if I make changes and then close the app, doesn't appears the warming "are you sure you want to close", or something similar.

**Reply by falkTX** [reply]
if you make changes you need to save the file again. there is no auto-save.

**Reply by Marco-56** [reply]
But I press the "save" button on FILE menu, and when I reopen Cardinal, changes are lost! May be it's a compatibility problem within Win11 and Cardinal 23.10!

**Reply by falkTX** [reply]
to be clear, you then used the file->open menu to open that same file again correct?

I think this might be related to non-ascii characters in the filename. try saving to something very simple like C:/test.vcv

**Reply by dromer** [reply]
Might it even be those characters somewhere else in the path?

**Reply by Marco-56** [reply]
I am testing with yours recommendations, thanks.

**Reply by subderground** [reply]
Hi, I'm on Win11 too and use vst3 version. The ''save'' function works fine.
Perhaps try redownload & install (manualy) Cardinal?

**Reply by Marco-56** [reply]
I finally found the answer: nothing about the characters, the guilty is the DISTRHO lldaeil Plugin host! In fact I downloaded Cardinal above all to use it in standalone mode, jamming with a piano sound with drones. So for first I've been looking for a plugin host to accomodate a VST2 piano plugin. I found DISTRHO lldaeil Plugin host that looks working fine for my resolutions (there is also DISTRHO Carla Plugin host, but I was unable to configure correctly, I also opened a topic on the Facebook Cardinal page, but nobody responded). And now, after a lot of test, I can say that if you just load that module and connect it to others in a patch, the SAVE function works; but if you also load a VST plugin in the DISTRHO lldaeil Plugin host (in "bridge" mode), the SAVE function works, but no others changes will be available for you; it means. the SAVE function doesn't works anymore!

**Reply by dromer** [reply]
There is no such Facebook page (and if it is then it's 100% unofficial and not a place to go for support).

I will try to reproduce this issue. Do you have it with any plugin loaded in Ildaeil module?

**Reply by dromer** [reply]
On Linux this works for me, I can make changes to my patch and save it:

![](https://i.imgur.com/c55JPwx.png)

It even saves the plugin state loaded into Ildaeil (in this case I have a VST2 loaded).

**Reply by Marco-56** [reply]
On Win 11 no, after you saved the patch with lldaeil, that will be your final patch; lldaeil remembers which plugin was loaded (also if I press the "reset" button), and no other changes looks possible. At the moment, the only way to make others changes looks do be deleting both lldaeil than his ExpanderMidiInput.

**Reply by falkTX** [reply]
might be then that the json file gets corrupted. can you upload here any patch that has this problem? (you might need to rename the file extension to be allowed on github upload, not sure)

**Reply by dromer** [reply]
Or put it in a zip file, that should be allowed.

**Comment by Marco-56** [comment]
[Marco_prova.zip](https://github.com/DISTRHO/Cardinal/files/14151242/Marco_prova.zip)
Et Voila! IIdaeil module is set to load MiniPiano by Air (a VST2) in bridged mode. There is also a DISTHRO file reader pointed to MP3 on my OneDrive; how I said, if I make any changes and then press SAVE on file menu, when I reopen that patch all changes are lost.

**Reply by dromer** [reply]
Might be the issue is with the OneDrive file though? It's very common for programs to have issues with reading/writing to such managed network drives.

**Reply by Marco-56** [reply]
No, I made a test with a patch without any link to OneDrive, and the result it's the same; when I load ildaeil Plugin Host AND I load a VST2 others changes aren't possible (if I don't load any on the Plugin Host, saving others changes is possible). May be the other module CARLA Plugin host, can make the same without problem, but I am unable to configure Carla Plugin Host (can you suggest me a good tutorial for this?).

**Reply by Marco-56** [reply]
Hi friends, What about the patch I uploaded? It works for you? Actually my only way to save changes on a patch with the ildaeil Plugin Host, is to delete it before saving (and then reload after the patch being loaded. May be CARLA plugin host can works better but I am unable to configure it. In particolar that module looks to be unable to find any VST on my computer (also if the path looks correct).

**Reply by dromer** [reply]
I don't have this plugin or audio file, but the patch loads yes:

You could try using the `load from file` option to point to a specific vst2/vst3/clap plugin directly (this doesn't work for lv2 at least):
![](https://i.imgur.com/Lv9g08p.png)

**Reply by Marco-56** [reply]
I tried the "load from file" option but the issue remains, But now I can be a little more precise; the problem is the "bridge mode". If I simply load a VST I will not have any "save changes" problems, BUT ildaeil Plugin, simply, doesn't play any sound! The VST looks loaded but his GUI doesn't respond to mouse actions and no sound at all! Otherwise, if I load a VST in bridge mode, the VST looks loaded too, but I can modified parameters on his GUI and, above all, it sounds! I really don't know what means "bridge mode" but it looks to be responsible of that issue.
